src/desktop/space/mod.rs

```diff
@@ -91,18 +87,20 @@ impl Space {
     /// If activate is true it will set the new windows state
     /// to be activate and removes that state from every
     /// other mapped window.
-    pub fn map_window<P, Z>(&mut self, window: &Window, location: P, z_index: Z, activate: bool)
+    pub fn map_element<P>(&mut self, element: E, location: P, activate: bool)
     where
         P: Into<Point<i32, Logical>>,
-        Z: Into<Option<u8>>,
     {
-        let z_index = z_index.into().unwrap_or(RenderZindex::Shell as u8);
-        {
-            let mut state = window_state(self.id, window);
-            state.location = location.into();
-            state.z_index = z_index;
+        if let Some(pos) = self.elements.iter().position(|inner| inner.element == element) {
+            self.elements.swap_remove(pos);
         }
-        self.insert_window(window, activate);
+
+        let inner = InnerElement {
+            element,
+            location: location.into(),
+            outputs: HashSet::new(),
```

Comment:
This does not seem right, `map_element` is also called during moving a window. So each move will reset the output list and after that `refresh` will always send an `enter`, but never a `leave`.

Reply:
fixed